src/marqo/tensor_search/tensor_search.py

```diff
@@ -364,13 +364,23 @@ def add_documents(config: Config, index_name: str, docs: List[dict], auto_refres
                                  'code': image_err.code})
                         )
                         break
-                
+
                 normalize_embeddings = index_info.index_settings[NsField.index_defaults][NsField.normalize_embeddings]
                 infer_if_image = index_info.index_settings[NsField.index_defaults][NsField.treat_urls_and_pointers_as_images]
+
+                try:
+                    model_properties = s2_inference.get_model_properties_from_registry(model_name=index_info.model_name)
```

Comment:
lets not do another call to s2_inference here, instead just check for the `model_properties` in the index defaults and pass through else pass through None